# Single-cell RNA-seq Simulated Datasets
Simulated single-cell RNA-seq datasets used in comparing different methods of differential expression analysis.

# Beta-Poisson simulation procedure
The simulated data set for isoform expression of single cells is generated by the beta-Poisson model (Vu,T.N. et al., 2016). In particular, we generate the counts for each isoform from a beta-Poisson distribution with four parameters estimated from a real data set (mESCs). The four-parameter beta-Poisson model is as follows

$BP_4(x|\alpha,\beta,\lambda_1,\lambda_2 ) = \lambda_2 \mbox{Poisson}(x|\lambda_1 \mbox{Beta}(\alpha,\beta)).$

The mean and variance of the model can be written as

$\mu = E(X)=\lambda_1\lambda_2 \phi_1$

and

$Var(X)=\mu \lambda_2 + \mu^2 \phi_2$,

where $\phi_1=\frac{\alpha}{\alpha+\beta}$ and 
$\phi_2=\frac{\beta}{\alpha(\alpha+\beta+1)}$. Crucially, we can modify the parameter $\lambda_1$ to create mean differences between groups. A more detailed description of the model can be referred to in the original study (Vu,T.N. et al., 2016).

Beta-Poisson models fitted on the real mESCs data set are used as baseline distributions for simulation. For each isoform, expression values across samples in the control and the treated group are generated from the same beta-Poisson model. To mimic the biological variation, 5\% of isoforms are selected to be differentially expressed between two groups (true DE isoforms). Specifically, the parameter $\lambda_1$, which controls the mean of the distribution, is fixed in the control group and multiplied by $\log_2$ fold change of 1 unit in the treated group. The effect direction is randomly determined for each DE isoform, with equal probability of up- and down-regulation. In other words, the quantity change between the two compared groups is either 2- or half-fold change with equal probability. The simulated data set consists of 80 samples in each of control and treated groups and a total of 10,000 isoforms measured per sample. Library sizes of the single-cell samples are randomly sampled from a range of 1-3 million. 
We filter out isoforms with zero expression across all samples. 

Related publication: 
Mou, T. et al. Reproducibility of methods to detect differentially expressed genes from  single-cell RNA sequencing.
Vu,T.N. et al. (2016) Beta-Poisson model for single-cell RNA-seq data analyses. Bioinformatics, btw202. http://bioinformatics.oxfordjournals.org/content/early/2016/04/18/bioinformatics.btw202
